doc/classes/WorkerThreadPool.xml

```diff
@@ -1,10 +1,50 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="WorkerThreadPool" inherits="Object" version="4.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
+		A singleton that allocates some [Thread]s on startup, used to offload tasks to these threads.
 	</brief_description>
 	<description>
+		The [WorkerThreadPool] singleton allocates a set of [Thread]s (called worker threads) on project startup and provides methods for offloading tasks to them. This can be used for simple multithreading without having to create [Thread]s.
+		Tasks hold the [Callable] to be run by the threads. [WorkerThreadPool] can be used to create regular tasks, which will be taken by one worker thread, or group tasks, which can be distributed between multiple worker threads. Group tasks execute the [Callable] multiple times, which makes them useful for iterating over a lot of elements, such as the enemies in an arena.
+		Here's a sample on how to offload an expensive function to worker threads:
+		[codeblocks]
+		[gdscript]
+		var enemies = [] # An array of enemies that must be processed every frame.
+
+		func process_enemy_ai(enemy_index):
+		    var processed_enemy = enemies[enemy_index]
+		    # Expensive logic...
+
+		func _process(delta):
+		    var task_id = WorkerThreadPool.add_group_task(process_enemy_ai, enemies.size())
+		    # Other code...
+		    WorkerThreadPool.wait_for_group_task_completion(task_id)
+		    # Other code that depends on the enemy AI already being processed.
+		[/gdscript]
+		[csharp]
+		private Node[] enemies = [] // An array of enemies that must be processed every frame.
+
+		private void processEnemyAI(int enemyIndex)
+		{
+		    Node processedEnemy = enemies[enemyIndex]
+		    // Expensive logic here.
+		}
+
+		public override void _Process(double delta)
+		{
+		    int taskID = WorkerThreadPool.addGroupTask(processEnemyAI, enemies.size())
+		    // Other code...
+		    WorkerThreadPool.waitForTaskCompletion(taskID)
+		    // Other code that depends on the enemy AI already being processed.
```

Comment:
The Godot C# API follows the .NET naming conventions so method names are PascalCase.

If using an array (`Node[]`) the equivalent to GDScript's `.size()` would be `.Length`. If using `Godot.Collections.Array` or `System.List`, the equivalent would be `.Count`.

Also, the method group can't be implicitly converted to a `Callable`, so we need to use the static `From` method.

```suggestion
		    long taskID = WorkerThreadPool.AddGroupTask(Callable.From&lt;int&gt;(ProcessEnemyAI), enemies.Length);
		    // Other code...
		    WorkerThreadPool.WaitForTaskCompletion(taskID);
		    // Other code that depends on the enemy AI already being processed.
```